ops/storage.py

```diff
@@ -220,6 +220,9 @@ def save_snapshot(self, handle_path: str, snapshot_data: Any) -> None:
             handle_path: The string identifying the snapshot.
             snapshot_data: The data to be persisted. (as returned by Object.snapshot()). This
                 might be a dict/tuple/int, but must only contain 'simple' python types.
+
+        Raises:
+            :class:`CalledProcessError`: if Juju returns an error.
```

Comment:
I don't think this one is particularly useful. The fact it uses `subprocess` is somewhat of an implementation detail, and users almost certainly won't be calling `save_snapshot` or `load_snapshot` in any case. Similar below.